src/effects/EffectManager.cpp

```diff
@@ -730,13 +730,39 @@ void EffectManager::SetBatchProcessing(const PluginID & ID, bool start)
 
 EffectUIHostInterface *EffectManager::GetEffect(const PluginID & ID)
 {
+   return DoGetEffect(ID).effect;
+}
+
+EffectSettings *EffectManager::GetDefaultSettings(const PluginID & ID)
+{
+   return GetEffectAndDefaultSettings(ID).second;
+}
+
+std::pair<EffectUIHostInterface *, EffectSettings *>
+EffectManager::GetEffectAndDefaultSettings(const PluginID & ID)
+{
+   auto &results = DoGetEffect(ID);
+   if (results.effect) {
+      auto &settings = results.settings;
+      if (!settings.has_value())
+         // Create on demand
+         settings = results.effect->GetDefinition().MakeSettings();
```

Comment:
I noticed that `RealtimeEffectState` uses `MakeSettings` directly to create it's own state (unique, supposedly), then the question is why do we need this function? I've searched for the uses of this function and turned out that it's indirectly used by `EffectManager::PromptUser` and `EffectUI::DoEffect` which, I suppose, should be parametrized with settings as some other functions in this PR. This two functions, in turn, are used in many places where 'default settings' may be supplied.

Reply:
If you were reviewing my commits sequentially, then you may written this before seeing that two later, I do that.  PromptUser is one function in which the default settings are retrieved from EffectManager.  The other is EffectUI::DialogFactory and DoEffect uses a pointer to that function.